var/spack/repos/builtin/packages/py-protobuf/package.py

```diff
@@ -71,7 +72,16 @@ class PyProtobuf(PythonPackage):
     depends_on("py-six@1.9:", when="@3:", type=("build", "run"))
     depends_on("py-ordereddict", when="@3: ^python@:2", type=("build", "run"))
     depends_on("py-unittest2", when="@3: ^python@:2", type=("build", "run"))
-    depends_on("protobuf", when="+cpp")
+
+    # Setup dependencies for protobuf to use the same minor version as py-protobuf
+    # Handle mapping the 4.x release to the protobuf 3.x releases
+    for ver in list(range(21, 22)):
+        depends_on("protobuf@3." + str(ver), type=("build", "run"), when="+cpp @4." + str(ver))
+    # Handle the 3.x series releases
+    for ver in list(range(1, 21)):
+        depends_on("protobuf@3." + str(ver), type=("build", "run"), when="+cpp @3." + str(ver))
+    # Handle the 2.x series releases
+    depends_on("protobuf@2", type=("build", "run"), when="+cpp @2")
```

Comment:
Out of curiosity, where did you find these mappings? 3 and 4 are somewhat intuitive, but is there official docs somewhere that state that any 2.x version is compatible with any 2.x version?

Reply:
@adamjstewart No I haven't found official documentation.  However, for the 2.x packages, there isn't a matching set of versions in both recipes.

Reply:
@adamjstewart There is a matching version so I tightened up the constraint.